src/net_processing.cpp

```diff
@@ -4132,7 +4160,9 @@ bool SendMessages(CNode* pto, CConnman& connman, const std::atomic<bool>& interr
 void AddPriorityDownload(const std::vector<const CBlockIndex*>& blocksToDownload, const PriorityDownloadCallback_t& callback) {
     LOCK(cs_main);
     for (const CBlockIndex* pindex: blocksToDownload) {
-        bool downloaded = pindex->nStatus & BLOCK_HAVE_DATA;
+        // We might have a valid block with no actual data, in the case of SPV filtering
+        // In that case we want to pretend it is downloaded, so we mark it as downloaded here
+        bool downloaded = (pindex->nStatus & BLOCK_HAVE_DATA) || ((pindex->nStatus & BLOCK_VALID_MASK) == BLOCK_VALID_MASK);
```

Comment:
Is BLOCK_VALID_MASK the best way to control this or can it cause other issue for SPV scanner?  I'm seeing lots of debug log warnings about prune height being triggered but am not sure if this is harmful or not.
Also there is some code that clears these flags for old blocks on wallet reload? (It won't prevent this from working right - but if its clearing the flag that may signal other potential issues)